src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/loading.ts

```diff
@@ -0,0 +1,39 @@
+// Can be used in various components to check whether an API call is waiting for a response for loading spinners
+// Example in Application.vue we do this in a button :loading="loadingStore.isLoading('application_post')"
+// Loading states are set by our apiResultHandler.ts where we provide a key based on the API operations we have ex. application_post
+
+import { defineStore } from "pinia";
+
+import type { OperationMethods } from "@/types/openapi";
+
+export type LoadingOperation = keyof OperationMethods;
+
+export type LoadingState = {
+  [key in LoadingOperation]: boolean;
+};
+
+export const useLoadingStore = defineStore("loading", {
+  state: (): LoadingState => ({
```

Comment:
Nice use of `OperationMethods` — wonder if we can get away from setting initial object of flags? I.e. LoadingState would be typed as you have it but initialized to `{}`, on get/set check `'{operation_method: OperationMethods}' in this` or something like that... Then this should just work as we add/update/remove endpoints.

Reply:
Yeah I think we can, let me play around. I think I need to do something with checking if a property exists then create it if it doesn't.